crates/compiler/build/src/link.rs

```diff
@@ -160,24 +154,24 @@ pub fn build_zig_host_native(
     // when we use zig 0.9. It looks like zig 0.10 is going to fix
     // this problem for us, so this is a temporary workaround
     if !target.contains("windows") {
-        command.args(&[
+        zig_cmd.args(&[
             // include the zig runtime
             "-fcompiler-rt",
         ]);
     }
 
     // valgrind does not yet support avx512 instructions, see #1963.
     if env::var("NO_AVX512").is_ok() {
-        command.args(&["-mcpu", "x86_64"]);
+        zig_cmd.args(&["-mcpu", "x86_64"]);
     }
 
     if matches!(opt_level, OptLevel::Optimize) {
-        command.args(&["-O", "ReleaseSafe"]);
+        zig_cmd.args(&["-O", "ReleaseSafe"]);
     } else if matches!(opt_level, OptLevel::Size) {
-        command.args(&["-O", "ReleaseSmall"]);
+        zig_cmd.args(&["-O", "ReleaseSmall"]);
     }
 
-    command.output().unwrap()
+    zig_cmd
```

Comment:
Returning the `Command` instead of the `Output` allows for more code re-use and better error message where the entire command with all args is printed.